### PR TITLE
Updates for Draft SoG/SJT Transfers

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#15795

*Description of changes:*
* Updates made for Draft Transfers for both Sale and Gift and Transfer to surviving joint tenants (Other Transfer types still wip, can revisit those when complete)
* Fixed issue causing draft resume for tranfer
* Updates to accommodate Reg Type inclusion in API response (Would try to resume DRAFT Mhr vs Transfer)
* Minor refactors.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue [hunk 1]
```diff
@@ -667,7 +667,7 @@ export default defineComponent({
         (isTransferToSurvivingJointTenant.value ? localState.isValidDeathCertificate : true) &&
         (isTransferToExecutorProbateWill.value ? TransWill.isValidTransfer.value : true)
       )
-    }, { deep: true })
+    }, { immediate: true, deep: true })
```
Comment (cameron-eyds) [inline ppr-ui/src/components/mhrRegistration/HomeOwners/HomeOwnersTable.vue:670]: Will do an initial validation
( To support Draft data being resumed if the user makes no changes after resuming )

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 2]
```diff
@@ -637,8 +643,6 @@ export default defineComponent({
 
         if (!mhrTransferFiling.error) {
           setUnsavedChanges(false)
-          // Delete the draft on successful submission
-          if (getMhrInformation.value.draftNumber) await deleteMhrDraft(getMhrInformation.value.draftNumber)
```
Comment (cameron-eyds) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue:641]: The API requires the `draftNumber` in the payload in order to DELETE the draft. 
We previously didn't include said draftNumber, so this was done in error. 

Now that we are including the draftNumber and following the spec, we no longer require this, the api does all the magic 👍

**Review by owillborn — APPROVED** [review]
LGTM, lots of logic

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (doug-lovett) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue:641]: Thanks for catching this.

**Review by doug-lovett — APPROVED** [review]
Looks great.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue:641]: No prob at all, it was my mistake, so easy to find 😆

**Pull request merged**